metadata/providers/BUILD

```diff
@@ -0,0 +1,16 @@
+exports_files(
```

Comment:
I'm pretty sure it shouldn't be necessary to duplicate the list of sources over two targets in this package. Looks like a nuisance to maintain. Did you experiment with removing one of them?

Also, what's the reasoning for avoiding the `bzl_library` rule? Users trying to generate their docs will find that a more convenient `dep` of their `bzl_library` targets

Reply:
We expect `@package_metadata` to be a dependency of virtually every other module, so we decided to not have any dependencies here ourselves.